x/storage/keeper/bucket_rate_limit.go

```diff
@@ -0,0 +1,290 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	paymenttypes "github.com/bnb-chain/greenfield/x/payment/types"
+	"github.com/bnb-chain/greenfield/x/storage/types"
+)
+
+func (k Keeper) SetBucketFlowRateLimit(ctx sdk.Context, operator sdk.AccAddress, bucketOwner sdk.AccAddress, paymentAccount sdk.AccAddress, bucketName string, rateLimit sdkmath.Int) error {
+	// check the operator is the same as the payment account owner
+	if !k.paymentKeeper.IsPaymentAccountOwner(ctx, paymentAccount, operator) {
+		return paymenttypes.ErrNotPaymentAccountOwner
+	}
+
+	// get the bucket
+	bucket, found := k.GetBucketInfo(ctx, bucketName)
+
+	// if the bucket does not use the payment account, just set the flow rate limit
+	if !found || (found && (bucket.PaymentAddress != paymentAccount.String() || bucket.Owner != bucketOwner.String())) {
+		// set the flow rate limit to the store
+		k.setBucketFlowRateLimit(ctx, paymentAccount, bucketOwner, bucketName, &types.BucketFlowRateLimit{
+			FlowRateLimit: rateLimit,
+		})
+		return nil
+	}
+
+	// set the flow rate limit for the bucket for the current bucket owner
+	err := k.setFlowRateLimit(ctx, bucket, paymentAccount, bucketName, rateLimit)
+	if err != nil {
+		return err
+	}
+
+	// set the flow rate limit to the store
+	k.setBucketFlowRateLimit(ctx, paymentAccount, bucketOwner, bucketName, &types.BucketFlowRateLimit{
```

Comment:
Do we allow mutliple rate limits (e.g., from different payment account)?
If not, we should check and prevent this.